internal/batches/service/service.go

```diff
@@ -57,30 +59,41 @@ const sourcegraphVersionQuery = `query SourcegraphVersion {
 
 // getSourcegraphVersion queries the Sourcegraph GraphQL API to get the
 // current version of the Sourcegraph instance.
-func (svc *Service) getSourcegraphVersion(ctx context.Context) (string, error) {
+
+func (svc *Service) getSourcegraphVersionAndMaxChangesetsCount(ctx context.Context) (string, int, error) {
 	var result struct {
-		Site struct {
+		maxUnlicensedChangesets int
+		Site                    struct {
 			ProductVersion string
 		}
 	}
 
-	ok, err := svc.client.NewQuery(sourcegraphVersionQuery).Do(ctx, &result)
+	ok, err := svc.client.NewQuery(getInstanceInfo).Do(ctx, &result)
 	if err != nil || !ok {
-		return "", err
+		return "", 0, err
 	}
 
-	return result.Site.ProductVersion, err
+	//add to return
+	return result.Site.ProductVersion, result.maxUnlicensedChangesets, err
 }
 
-// DetermineFeatureFlags fetches the version of the configured Sourcegraph and
+// DetermineFeatureFlagsAndLicense fetches the version of the configured Sourcegraph and
 // returns the enabled features.
-func (svc *Service) DetermineFeatureFlags(ctx context.Context) (*batches.FeatureFlags, error) {
-	version, err := svc.getSourcegraphVersion(ctx)
+
+func (svc *Service) DetermineFeatureFlagsAndLicense(ctx context.Context) (*batches.LicenseRestrictions, *batches.FeatureFlags, error) {
+	version, mc, err := svc.getSourcegraphVersionAndMaxChangesetsCount(ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to query Sourcegraph version to check for available features")
+		return nil, nil, errors.Wrap(err, "failed to query Sourcegraph version and license info for instance capabilities")
+	}
+
+	lr := &batches.LicenseRestrictions{
+		MaxUnlicensedChangesets: mc,
 	}
+
 	ffs := &batches.FeatureFlags{}
-	return ffs, ffs.SetFromVersion(version)
+
+	return lr, ffs, ffs.SetFromVersion(version)
+
```

Comment:
It's so spacious in here lol 😜

```suggestion
	return lr, ffs, ffs.SetFromVersion(version)
```